File: ICPC WF 2021 Dhaka Practice /cses_1095.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
ll mod = 1000000007;

ll power(ll r, ll n){
    if (n == 0)
        return 1;
    if (n%2 == 0){
        ll v = power(r, n/2);
        return v*v % mod;
    }
    else
        return ((power(r, n-1) % mod) * r) % mod;
}
int main(){
    ll t, a, b;
    cin>>t;
    while(t--){
        cin>>a>>b;
        if (a+b == 0)cout<<1<<endl;
        else {
            cout <<power(a,b) << endl;
        }
    }
}

File: ICPC WF 2021 Dhaka Practice /cses_1095.py
mod = int(1e9) + 7
def power(r, n):
    if n == 0:
        return 1
    if n%2 == 0:
        v = power(r, n//2)
        return v*v % mod
    else:
        return (power(r, n-1) * r) % mod
    
for _ in range(int(input())):
    a,b = map(int, input().split())
    if a+b == 0:
        print(1)
    else:
        print(power(a,b))
        
